tests/phpunit/tests/user.php

```diff
@@ -915,6 +915,78 @@ public function test_wp_insert_user_should_reject_user_login_that_matches_existi
 		$this->assertSame( 'existing_user_email_as_login', $user_id->get_error_code() );
 	}
 
+	/**
+	 * Tests that `wp_update_user()` allows updating a user when
+	 * the `user_login` and `user_email` are the same.
+	 *
+	 * @ticket 57967
+	 *
+	 * @covers ::wp_update_user
```

Comment:
Pinging @hellofromtonya for a confidence check.

This PR only touches `wp_insert_user()`. It may therefore be necessary to either add `@covers ::wp_insert_user` to the covered functions, or change the test to only target the behaviour of `wp_insert_user()` (i.e. no calls to `wp_update_user()`).

Not 100% sure on the right call here.

Reply:
Thanks for the ping @costdev.

The Trac ticket specifically mentions using `wp_update_user()`. While the PR touches only the `wp_insert_user()`, using `wp_update_user()` helps to document the issue.

So I think adding an `@covers` for both functions is needed.